scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -434,9 +444,12 @@
   "Update the cook expected state. Include some business logic to e.g., not change a state to the same value more than once. Marks any state changes Also has a lattice of state. Called externally and from state machine."
   [{:keys [cook-expected-state-map] :as compute-cluster} pod-name new-cook-expected-state-dict]
   (locking (calculate-lock [pod-name])
-    (let [old-state (get @cook-expected-state-map pod-name)]
-      (when-not (cook-expected-state-equivalent? new-cook-expected-state-dict old-state)
-        (swap! cook-expected-state-map assoc pod-name new-cook-expected-state-dict)
+    (let [old-state (get @cook-expected-state-map pod-name)
+          ; Save the launch pod. We may need it in order to kill it. See note under (:killed, :missing) in process, above.
+          old-pod (:launch-pod old-state)
+          new-expected-state-dict-merged (merge {:launch-pod old-pod} new-cook-expected-state-dict)]
```

Comment:
Can we `assoc` here instead of `merge`-ing?

Reply:
Not obviously. What I want here is to use the entire new-cook-expected-state-dict, except that if it lacks :launch-pod, we should copy from the prior state, if it exists, without having to enumerate the specific keys in new-cook-expected-state-dict. I can't think of a nice idiom to do this with assoc as it overwrites whats already there.

Reply:
Ah, makes sense, thanks